protocol/src/main/scala/io/mediachain/protocol/CborSerialization.scala

```diff
@@ -601,6 +606,47 @@ object CborSerialization {
         )
   }
 
+  object JournalBlockArchiveDeserializer extends CborDeserializer[JournalBlockArchive]
+  {
+    import scala.annotation.tailrec
+
+    def fromCMap(cMap: CMap): Xor[DeserializationError, JournalBlockArchive] =
+      for {
+        _ <- assertRequiredTypeName(cMap, MediachainTypes.JournalBlockArchive)
+        ref <- getRequiredReference(cMap, "ref")
+        xblock <- getRequired[CMap](cMap, "block")
+        block <- JournalBlockDeserializer.fromCMap(xblock)
+        xdata <- getRequired[CArray](cMap, "data")
+        data <- dataFromCbor(xdata)
+      } yield
+        JournalBlockArchive(
+          ref = ref,
+          block = block,
+          data = data
+        )
+    
+    def dataFromCbor(records: CArray) = {
+      @tailrec def loop(rest: List[CValue], map: Map[Reference, Array[Byte]])
```

Comment:
hmm you're right using flatMap breaks the @tailrec, I'd have assumed that the two statements would get compiled to equivalent bytecode :\